### PR TITLE
Add a synoptic map example

### PR DESCRIPTION
**Pull request by Cadair**
This is a werid one, I wrote it today as a bit of an experiment.

I am not really sure what this example is showing. Parsing a unsupported file, creating a custom coordinate frame, doing crazy things with WCSAxes.

Opinions on if this should go in?

### CODE REVIEW

**Comment by Cadair** [comment]
ping @larrymanley 

@mbobra This is based on your IDL code.

**Comment by Cadair** [comment]
oh, a Gist of the notebook version is here: https://gist.github.com/Cadair/ba1bd6f17caf880f0f5122fb61b9739d

File: examples/gallery/hmi_synoptic_maps.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    sin(deg) to the real HGC frame. It is defined backwards because the WCS
+    object for the HGC sin(deg) frame reads in as a proper HGC frame due to the
+    values of CTYPE. This frame therefore converts from what WCS thinks is HGC
+    to ASineHGC but is in reality converting from ASineHGC to HGC.
+    """
+
+
+@frame_transform_graph.transform(coord.FunctionTransform,
+                                 spc.HeliographicCarrington, ASineHGC)
+def hgc_to_sinehgc(hgc_coord, sinehgc_frame):
```
Comment (dpshelio) [inline examples/gallery/hmi_synoptic_maps.py]: Should this (and below) be indented?

Comment (Cadair) [inline examples/gallery/hmi_synoptic_maps.py]: No it's not part of the class. The class body is empty.

**Comment by Cadair** [comment]
Note to self, from the communtiy meeting.

This needs a little tidy up to make it a tutorial about reading in unsupported files and handling unsupported coordinate systems. So more description and critially a better title.

**Pull request merged**